src/Microsoft.OData.Edm/ExtensionMethods/EdmTypeSemantics.cs

```diff
@@ -1216,18 +1221,33 @@ public static EdmPrimitiveTypeKind PrimitiveKind(this IEdmTypeReference type)
         /// <param name="potentialBaseType">The potential base type of the type being tested.</param>
         /// <returns>True if and only if the type inherits from the potential base type.</returns>
         public static bool InheritsFrom(this IEdmStructuredType type, IEdmStructuredType potentialBaseType)
+        {
+            HashSet<string> baseTypes;
+            string fullName = type.FullTypeName();
+
+            if (!baseTypeCache.TryGetValue(fullName, out baseTypes))
+            {
+                baseTypes = new HashSet<string>();

```

Comment:
why pass in the HashSet rather than creating it in GetBaseTypes and returning it?